### PR TITLE
[Doc] Add round robin sharding documentation

### PR DESCRIPTION
**Pull request by NivekT**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #1050

Also fixed some wording and typos

Differential Revision: [D43625322](https://our.internmc.facebook.com/intern/diff/D43625322)

### CODE REVIEW

**Review by ejguan — APPROVED** [review]
This looks awesome. Thank you so much. Let's cherrypick it before the final branch cut.

File: docs/source/reading_service.rst [hunk 1]
```diff
@@ -13,9 +15,9 @@ Dynamic Sharding
...
+- ``sharding_filter`` (:class:`ShardingFilter`): When the pipeline is replicable, each distributed/multiprocessing worker loads data from its own replica of the ``DataPipe`` graph, while skipping samples that do not belong to the corresponding worker at the point where ``sharding_filter`` is placed.
 
-- ``sharding_round_robin_dispatch``: When there is any ``sharding_round_robin_dispatch`` ``DataPipe`` in the pipeline, that branch will be treated as a non-replicable branch. Then, a single dispatching process will be created to load data from the non-repliable branch and distributed data to the subsequent worker processes.
+- ``sharding_round_robin_dispatch`` (:class:`ShardingRoundRobinDispatcher`): When there is any ``sharding_round_robin_dispatch`` ``DataPipe`` in the pipeline, that branch (i.e. all DataPipes prior to ``sharding_round_robin_dispatch``) will be treated as a non-replicable branch. A single dispatching process will be created to load data from the non-replicable branch and distributed data to the subsequent worker processes.
```
[the remark this answers is not in the document]
Comment (NivekT) [reply docs/source/reading_service.rst:20]: Verified

File: docs/source/reading_service.rst [hunk 2]
```diff
@@ -13,9 +15,9 @@ Dynamic Sharding
 
 Dynamic sharding is achieved by ``MultiProcessingReadingService`` and ``DistributedReadingService`` to shard the pipeline based on the information of corresponding multiprocessing and distributed workers. And, TorchData offers two types of ``DataPipe`` letting users to define the sharding place within the pipeline.
 
-- ``sharding_filter``: When the pipeline is replicable, each distributed/multiprocessing worker loads data from one replica of the ``DataPipe`` graph, and skip the data not blonged to the corresponding worker at the place of ``sharding_filter``.
+- ``sharding_filter`` (:class:`ShardingFilter`): When the pipeline is replicable, each distributed/multiprocessing worker loads data from its own replica of the ``DataPipe`` graph, while skipping samples that do not belong to the corresponding worker at the point where ``sharding_filter`` is placed.
```
[the remark this answers is not in the document]
Comment (NivekT) [reply docs/source/reading_service.rst:18]: Verified

**Comment by NivekT** [comment]
@nivekt has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D43625322).

**Pull request closed without merging**